src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb

```diff
@@ -149,7 +149,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Retargeting
             Private Function RetargetNamedTypeDefinition(type As NamedTypeSymbol, options As RetargetOptions) As NamedTypeSymbol
                 Debug.Assert(type Is type.OriginalDefinition)
 
-                ' Before we do anything else, check if we need to do special retargeting
+                If (type.IsTupleType) Then
+                    Return DirectCast(type, TupleTypeSymbol).WithUnderlyingType(Retarget(type.TupleUnderlyingType, options))
```

Comment:
The C# version has an extra check of `newUnderlyingType.IsTupleOrCompatibleWithTupleOfCardinality`. When does that fail and is the same check needed here?